### DISCUSSION TITLE
__root__ -> items -> 0 -> datasetName   field required (type=value_error.missing)

### DISCUSSION THREAD

**Discussion by emailic** [langfuse/langfuse#2154] [Support]
Hello! I just upgraded Langfuse using poetry. ( `poetry update langfuse` in the appropriate directory).

however, for some reason, our datasets are not being read correctly anymore.

Namely:

```
from langfuse import Langfuse
import os

langfuse = Langfuse(public_key= os.environ["LANGFUSE_EVALUATION_PUBLIC_KEY"], secret_key= os.environ["LANGFUSE_EVALUATION_SECRET_KEY"],  debug=True)

dataset = langfuse.get_dataset("mini_esg_bench")
```

The above script results in this error by pydantic:

```
(noa-py3.11) email@MacBook-Pro-3 noa-chat % python jobs/evaluation/test_langfuse.py
DEBUG:langfuse:consumer is running...
DEBUG:langfuse:Getting datasets mini_esg_bench
ERROR:langfuse:50 validation errors for ParsingModel[Dataset]
__root__ -> items -> 0 -> datasetName
  field required (type=value_error.missing)
__root__ -> items -> 1 -> datasetName
  field required (type=value_error.missing)
__root__ -> items -> 2 -> datasetName
  field required (type=value_error.missing)
...
__root__ -> items -> 49 -> datasetName
  field required (type=value_error.missing)
Traceback (most recent call last):
  File "/Users/email/Library/Caches/pypoetry/virtualenvs/noa-77-C5xh0-py3.11/lib/python3.11/site-packages/langfuse/client.py", line 267, in get_dataset
    dataset = self.client.datasets.get(dataset_name=name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/email/Library/Caches/pypoetry/virtualenvs/noa-77-C5xh0-py3.11/lib/python3.11/site-packages/langfuse/api/resources/datasets/client.py", line 187, in get
    return pydantic_v1.parse_obj_as(Dataset, _response.json())  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pydantic/tools.py", line 38, in pydantic.tools.parse_obj_as
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 50 validation errors for ParsingModel[Dataset]
__root__ -> items -> 0 -> datasetName
  field required (type=value_error.missing)
__root__ -> items -> 1 -> datasetName
  field required (type=value_error.missing)
__root__ -> items -> 2 -> datasetName
  field required (type=value_error.missing)
...
__root__ -> items -> 49 -> datasetName
  field required (type=value_error.missing)
Traceback (most recent call last):
  File "/Users/email/Documents/Chaptr-LAPTOP-LTENNM0H/github/noa-chat/jobs/evaluation/test_langfuse.py", line 7, in <module>
    dataset = langfuse.get_dataset("mini_esg_bench")
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/email/Library/Caches/pypoetry/virtualenvs/noa-77-C5xh0-py3.11/lib/python3.11/site-packages/langfuse/client.py", line 274, in get_dataset
    raise e
  File "/Users/email/Library/Caches/pypoetry/virtualenvs/noa-77-C5xh0-py3.11/lib/python3.11/site-packages/langfuse/client.py", line 267, in get_dataset
    dataset = self.client.datasets.get(dataset_name=name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/email/Library/Caches/pypoetry/virtualenvs/noa-77-C5xh0-py3.11/lib/python3.11/site-packages/langfuse/api/resources/datasets/client.py", line 187, in get
    return pydantic_v1.parse_obj_as(Dataset, _response.json())  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pydantic/tools.py", line 38, in pydantic.tools.parse_obj_as
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 50 validation errors for ParsingModel[Dataset]
__root__ -> items -> 0 -> datasetName
  field required (type=value_error.missing)
__root__ -> items -> 1 -> datasetName
  field required (type=value_error.missing)
...
  field required (type=value_error.missing)
__root__ -> items -> 49 -> datasetName
  field required (type=value_error.missing)
DEBUG:langfuse:joining 1 consumer threads
DEBUG:langfuse:~0 items in the Langfuse queue
DEBUG:langfuse:consumer thread 0 joined

```

Bear in mind:
- we didn't change our datasets since restarting langfuse.
- This was tested on 2 of our datasets and they are both failing
- Langfuse credentials are correct and haven't changed
- datasets appear ok and haven't changed in a long time (image of one item below)
![image](https://github.com/langfuse/langfuse/assets/83206159/264bc8e2-7e7d-44d2-9bf5-942c67d6f154)

To fix this, we tried:

`cd directory/where/the/script/is`

`Poetry install
`
`poetry shell
`
`poetry export -f requirements.txt --output requirements.txt
`
`pip install -r requirements.txt
`

Any idea on what is causing Langfuse datasets not to load correctly?

**Comment by marcklingen** [comment]
thanks for raising this, I am currently having a look

**Comment by marcklingen** [comment]
Your code looks good, I've just tried to reproduce this locally, but failed (see [test](https://github.com/langfuse/langfuse-python/pull/708/files) added to python sdk repo, passes locally).

**Reply by emailic** [reply]
Thank you @marcklingen , in that case I'm assuming it must be something from our side, whether poetry related or local setup. Will keep on looking into this issue and will let u know once it's resolved, and how we resolved it. Thank you once again and take care, Ema

**Reply by marcklingen** [reply]
sure, let me know in case you cannot resolve this. If there is any chance that you can help me reproduce, I'm happy to look into this

**Reply by emailic** [reply]
Hey Marc, we are now facing this same problem when moving from Langfuse version 2.26 to newer versions (tested 2.31 and 2.33) in the other team as well.

Since the error is from pydantic, mind letting us know which version of Pydantic and Json are you using?

**Reply by hassiebp** [reply]
Hi @emailic - this is most probably due to a missing field `datasetName` in the dataset items returned from the API. We have added this [a while back](https://github.com/langfuse/langfuse/commit/bd98d20f6573944bf31324ce14d86f5c83575363#diff-c167852b3a778bb45b3b604cb5e53ab0252c61a1fa38b4f3a6f14dce3203e194). Are you running a self-hosted instance or are you using Langfuse cloud? If you are self-hosted, please upgrade your Langfuse instance. If you are on Langfuse cloud, please share your project ID referencing this issue via email through support at langfuse.com for further debugging :)
